cmd/controller/main.go

```diff
@@ -211,14 +216,44 @@ func main() {
 		}
 	}
 
-	startUtilHttpServerOrDie(kubeClient, kubeconfigProvider, stopCh)
-	startControllerViaLeaderElectionOrDie(runOptions)
+	ctx, ctxDone := context.WithCancel(context.Background())
+	runOptions.ctx = ctx
+	var g run.Group
+	{
+		g.Add(func() error {
+			return startUtilHttpServer(kubeClient, kubeconfigProvider, ctx)
```

Comment:
Take `startUtilHttpServer` out of that function and create the `http.Server{}` directly before adding this group. Then there's no need to pass the context. ☺️

Reply:
I think that having a separate function that encloses and manages creation of an HTTP server is actually nice and readable. In addition I also think that passing a context to dependent function is actually okay as context was designed for that purpose ? (To propagate a cancellation).